proposals/semi-auto-properties.md

```diff
@@ -10,6 +10,228 @@ Two common scenarios are that you want to apply a constraint on the setter, ensu
 
 In these cases by now you always have to create an instance field and write the whole property yourself.  This not only adds a fair amount of code, but it also leaks the `field` into the rest of the type's scope, when it is often desirable to only have it be available to the bodies of the accessors.
 
+## Detailed design
+
+For properties with an `init` accessor, everything that applies below to `set` would apply instead to the `init` accessor.
+
+**Principle 1:** Every property can be thought of as having a backing field by default, which is elided when not used. The field is referenced using the keyword `field` and its visibility is scoped to the accessor bodies.
```

Comment:
Consider discussing how `override` plays into this. For example consider the following: 

```csharp
public override string Prop { get => field ?? "default" };
```

In this case does `field` represent a value on the current type or am I accessing the value in the base type? Essentially does every declaration of a property have a backing field or is it only when there is a new slot?

Reply:
I think the design should align with precedent for existing auto-properties, where declaring an override auto-property adds a field to the containing type (in which the override is declared).

Derived types don't have knowledge of whether a property on a base type is auto-implemented. I think we should keep it that way for this feature as well.

Reply:
> I think the design should align with precedent for existing auto-properties, where declaring an override auto-property adds a field to the containing type (in which the override is declared).

That seems like a fine interpretation but I think we should make that very explicit in the proposal here. 

Now ... same question but for explicit interface implementations.

Reply:
Thanks, updated!